pkg/nodes/node.go

```diff
@@ -23,10 +25,77 @@ const (
 
 // Builder provides struct for Node object containing connection to the cluster and the list of Node definitions.
 type Builder struct {
-	Definition *v1.Node
-	Object     *v1.Node
-	apiClient  *clients.Settings
-	errorMsg   string
+	Definition  *v1.Node
+	Object      *v1.Node
+	apiClient   *clients.Settings
+	errorMsg    string
+	drainHelper *drain.Helper
+}
+
+// NewDrainHelper builds drain Helper that contains parameters to control the behaviour of drain.
+func (builder *Builder) NewDrainHelper(
+	force bool,
+	ignoreDaemonsets bool,
+	gracePeriod int,
+	timeout time.Duration,
+	skipWaitForDeleteTimeoutSeconds int,
+	deleteLocalData bool) *Builder {
+	glog.V(100).Infof("Creating new DrainOptions config")
+	glog.V(100).Infof(
+		fmt.Sprintf("Setting DrainOption 'force': %v", force))
+	glog.V(100).Infof(
+		fmt.Sprintf("Setting DrainOption 'gracePeriod': %d seconds", gracePeriod))
+	glog.V(100).Infof(
+		fmt.Sprintf("Setting DrainOption 'skipWaitForDeletionTimeout': %d seconds", skipWaitForDeleteTimeoutSeconds))
+	glog.V(100).Infof(
+		fmt.Sprintf("Setting DrainOption 'ignoreAllDaemonSets': %v", ignoreDaemonsets))
+	glog.V(100).Infof(
+		fmt.Sprintf("Setting DrainOption 'timeout': %v", timeout))
+	glog.V(100).Infof(
+		fmt.Sprintf("Setting DrainOption 'deleteEmptyDir': %v", deleteLocalData))
+
+	builder.drainHelper = &drain.Helper{
+		Ctx:    context.TODO(),
+		Client: builder.apiClient.K8sClient,
+		// Delete pods that do not declare a controller.
+		Force: force,
+		// GracePeriodSeconds is how long to wait for a pod to terminate.
+		GracePeriodSeconds: gracePeriod,
+		// Ignore DaemonSet-managed pods
+		IgnoreAllDaemonSets: ignoreDaemonsets,
+		// The length of time to wait before giving up
+		Timeout: timeout,
+		// Local data from emptyDir volumes will be deleted
+		DeleteEmptyDirData: deleteLocalData,
+		// when the node is drained
```

Comment:
need a couple more words. Not clear

Reply:
bad split, that's related to the previous line, updated